Stratis.Bitcoin.Tests/Features/MemoryPool/TestChainFactory.cs

```diff
@@ -0,0 +1,221 @@
+﻿using Microsoft.Extensions.Logging;
+using Moq;
+using NBitcoin;
+using NBitcoin.Protocol;
+using Stratis.Bitcoin.Base;
+using Stratis.Bitcoin.BlockPulling;
+using Stratis.Bitcoin.Configuration;
+using Stratis.Bitcoin.Connection;
+using Stratis.Bitcoin.Features.Consensus;
+using Stratis.Bitcoin.Features.Consensus.CoinViews;
+using Stratis.Bitcoin.Features.Consensus.Deployments;
+using Stratis.Bitcoin.Features.MemoryPool;
+using Stratis.Bitcoin.Features.MemoryPool.Fee;
+using Stratis.Bitcoin.Features.Miner;
+using Stratis.Bitcoin.Utilities;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Stratis.Bitcoin.Tests.Features.MemoryPool
+{
+    /// <summary>
+    /// Public interface for test chain context.
+    /// </summary>
+    public interface ITestChainContext
+    {
+        /// <summary>
+        /// Memory pool validator interface;
+        /// </summary>
+        IMempoolValidator MempoolValidator { get;  }
+
+        /// <summary>
+        /// List of the source transactions in the test chain.
+        /// </summary>
+        List<Transaction> SrcTxs { get; }
+    }
+
+    /// <summary>
+    /// Concrete instance of the test chain.
+    /// </summary>
+    internal class TestChainContext : ITestChainContext
+    {
+        /// <inheritdoc />
+        public IMempoolValidator MempoolValidator { get; set; }
+
+        /// <inheritdoc />
+        public List<Transaction> SrcTxs { get; set; }
+    }
+
+   /// <summary>
+   /// Factory for creating the test chain.
+   /// Much of this logic was taken directly from the embedded TestContext class in MinerTest.cs in the integration tests.
+   /// </summary>
+    internal class TestChainFactory
+    {
+        /// <summary>
+        /// Creates the test chain with some default blocks and txs.
+        /// </summary>
+        /// <param name="network">Network to create the chain on.</param>
+        /// <param name="scriptPubKey">Public key to create blocks/txs with.</param>
+        /// <returns>Context object representing the test chain.</returns>
+        public static ITestChainContext Create(Network network, Script scriptPubKey)
+        {
+            NodeSettings nodeSettings = NodeSettings.Default();
```

Comment:
You probably want to specify a test folder (is the folder hit?)

Reply:
Do that in the next PR

Reply:
Will do! Thanks.